### PR TITLE
Feature/jmll 1166 pay cut off date

### PR DESCRIPTION
**Pull request by Luisella21**

### CODE REVIEW

File: leavers/utils/workday_calculation.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+from datetime import date, timedelta
+from govuk_bank_holidays.bank_holidays import BankHolidays
+
+PAY_CUT_OFF_DAY = 3  # The third of the month is the designated cutoff date
+ALTERNATIVE_WEEK_DAY_DESIGNATED = 4  # Friday
+PAY_CUT_OFF_INTERVAL = 4
+
+
+def working_day_delta(start_date: date, working_day_delta: int) -> date:
```
Comment (jafacakes2011) [inline leavers/utils/workday_calculation.py:9]: Please can we get a comment that explains what this function's purpose is. Will be super helpful for any new dev :)

File: leavers/utils/workday_calculation.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
+from datetime import date, timedelta
+from govuk_bank_holidays.bank_holidays import BankHolidays
+
+PAY_CUT_OFF_DAY = 3  # The third of the month is the designated cutoff date
+ALTERNATIVE_WEEK_DAY_DESIGNATED = 4  # Friday
+PAY_CUT_OFF_INTERVAL = 4
```
Comment (jafacakes2011) [inline leavers/utils/workday_calculation.py:6]: What does this interval represent?
is it the amount of days before payroll cut off that we would start to notify HR? if so, I think we might need a comment

File: leavers/utils/workday_calculation.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+    # Find the designated day, in this case the previous Friday
+    cut_off_day_of_week = cutoff_day.weekday()
+    delta = cut_off_day_of_week - ALTERNATIVE_WEEK_DAY_DESIGNATED
+    if delta <= 0:
+        delta += 7
+    cutoff_day -= timedelta(delta)
+    # If the previous designated day is also a bankholiday,
+    # return the previous working day
+    while not bank_holidays.is_work_day(cutoff_day, division=area):
```
Comment (jafacakes2011) [inline leavers/utils/workday_calculation.py:43]: Really like this `while`!

**Review by jafacakes2011 — COMMENTED** [review]
Some great work here!
I'd love to see some detailed tests for these methods

File: leavers/utils/workday_calculation.py (hunk 3 above)
Comment (Luisella21) [reply leavers/utils/workday_calculation.py:68]: Thank you!

**Comment by jafacakes2011** [comment]
Closing this as it is old

**Pull request closed without merging**